packages/ra-ui-materialui/src/list/filter/FilterListItem.tsx

```diff
@@ -144,27 +145,30 @@ const useStyles = makeStyles(theme => ({
  */
 const FilterListItem: FC<{ label: string; value: any }> = props => {
     const { label, value } = props;
-    const { filterValues, setFilters } = useListFilterContext();
+    const {
+        filterValues,
+        setFilters,
+        displayedFilters,
+        hideFilter,
+    } = useListContext();
     const translate = useTranslate();
     const classes = useStyles(props);
 
-    const isSelected = isEqual(value, filterValues);
+    const isSelected =
+        isEmpty(value) || isEmpty(filterValues)
+            ? false
+            : isMatch(filterValues, value);
 
     const addFilter = () => {
-        setFilters({ ...filterValues, ...value }, null, false);
+        setFilters(merge({}, filterValues, value), null, false);
     };
 
     const removeFilter = () => {
-        const keysToRemove = Object.keys(value);
-        const filters = Object.keys(filterValues).reduce(
-            (acc, key) =>
-                keysToRemove.includes(key)
-                    ? acc
-                    : { ...acc, [key]: filterValues[key] },
-            {}
-        );
-
-        setFilters(filters, null, false);
+        setFilters(omit(filterValues, paths(value)), null, false);
+        const displayedFilterToRemove = Object.keys(
+            displayedFilters
+        )?.find(elem => paths(value).includes(elem));
+        hideFilter(displayedFilterToRemove);
```

Comment:
Problem: We found out that the displayedFilter key of the URL was not properly cleaned when remove a sidebar filter, therefore leave standard Filters shown.
Correction: manual cleaning